tests/test_client.py

```diff
@@ -147,3 +153,225 @@ def on_message(client, userdata, msg):
 
         packet_in = fake_broker.receive_packet(1)
         assert not packet_in  # Check connection is closed
+
+
```

Comment:
How many of these tests could be replaced by tests on the `WebsocketWrapper` class directly, using a fake socket object? That should be much easier than creating a client and having it talk to a fake broker.